### PR TITLE
IA-2946 fix tests

### PR DESCRIPTION
**Pull request by quang-le**
Explain what problem this PR is resolving

Related JIRA tickets : IA-2946

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc
Nope

## Changes

- Add `withRouter` to enzyme tests
- Convert tests that still failed to cypress
- Delete obsolete enzyme tests, ie: those converted to cypress and those that only checked if page rendered (which is checked by default in cypress)
- Add `dataTestId` prop to `InputComponent` (requires blsq-comp update)
- Add missingtranslation in `UserInfos`
- Updated urls in some tests to add new params in outdated urls (eg: when new filters have been added since the test was first written)

## How to test

run enzyme and cypress tests

Passing cypress tests: https://github.com/BLSQ/iaso/actions/runs/9171714830
related commit: https://github.com/BLSQ/iaso/pull/1309/commits/5838a7137dffee39e51a17510fb21e6ccbe162f5

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/forms/config/index.spec.js [hunk 1]
```diff
@@ -41,24 +40,17 @@ describe('Forms config', () => {
...
-            expect(openStub).to.have.been.called;
-            sinon.restore();
-        });
+        //         expect(xlsButton).to.have.lengthOf(1);
+        //         const openStub = sinon.stub(window, 'open');
+        //         xlsButton.props().onClick();
+        //         expect(openStub).to.have.been.called;
+        //         sinon.restore();
+        //     });
     });
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/forms/config/index.spec.js]: you can skip it instead, so we still have a trace while running tests

File: hat/assets/js/cypress/integration/02 - forms/details.spec.js [hunk 2]
```diff
@@ -26,28 +29,139 @@ const formVersionsPageOne = makePaginatedResponse({
...
-    describe.skip('Filters', () => {});
+    describe('When updating', () => {
+        it('Save button is disabled if no project', () => {
+            cy.wait(['@getForm', '@getFormVersions']);
+            // cy.get('#project_ids').as('projectsInput').click();
+            // cy.get('@projectsInput')
+            //     .parent()
+            //     .within(() => {
+            //         cy.get('[data-testid="ClearIcon"]').click();
+            //     });
```
Comment (beygorghor) [inline hat/assets/js/cypress/integration/02 - forms/details.spec.js]: 🗑️  ?

File: hat/assets/js/cypress/integration/02 - forms/list.spec.js [hunk 3]
```diff
@@ -4,53 +4,55 @@ import listFixture from '../../fixtures/forms/list.json';
...
+    //     times: 2,
+    // };
+    // cy.intercept({ ...options, query: formQuery }, req => {
+    //     req.on('response', response => {
+    //         if (response.statusMessage === 'OK') {
+    //             interceptFlag = true;
+    //             response.send({ fixture });
+    //         }
+    //     });
+    // }).as('getFormsWithQuery');
```
Comment (beygorghor) [inline hat/assets/js/cypress/integration/02 - forms/list.spec.js]: 🗑️  ?

File: hat/assets/js/cypress/support/commands.js [hunk 4]
```diff
@@ -93,6 +94,11 @@ Cypress.Commands.add(
  * @param {number} newOuIndex - index of the new selected ou
  */
 Cypress.Commands.add('fillTreeView', (id, newOuIndex, clear = true) => {
+    cy.intercept(
+        'GET',
+        '/api/orgunits/treesearch/?&rootsForUser=true&defaultVersion=true&validation_status=VALID&ignoreEmptyNames=true',
+        orgUnits,
+    ).as('getTreeview');
```
Comment (beygorghor) [inline hat/assets/js/cypress/support/commands.js]: ❤️

**Review by beygorghor — APPROVED** [review]

File: hat/assets/js/cypress/integration/02 - forms/details.spec.js (hunk 2 above)
Comment (quang-le) [reply hat/assets/js/cypress/integration/02 - forms/details.spec.js]: Yes, the dropdown is selection is already empty, so trying to find the clear button will cause the test to fail

**Pull request merged**